### PR TITLE
Add logging relation

### PR DESCRIPTION
**Pull request by rgildein**
Add logging relation and use the Loki URL in spark properties generated by integration-hub. This feature requires to be used with latest 3.4.2 charmed-spark image, otherwise it will not work.

Related with: https://github.com/canonical/charmed-spark-rock/pull/114

How I tested it:
```bash
$ juju add-model spark
$ juju deploy spark-k8s-bundle --channel 3.4/edge --trust
$ charmcraft pack
$ juju refresh integration-hub --path ./spark-integration-hub-k8s_ubuntu-22.04-amd64.charm --force-units
# configure mino
$ juju consume u2-k8s:cos.loki-logging
$ juju deploy grafana-agent-k8s --channel=stable
$ juju integrate grafana-agent-k8s loki-logging
$ juju integrate grafana-agent-k8s integration-hub:logging

# check the configuration provided by integration-hub
$ spark-client.service-account-registry create --username spark --namespace spark
$ spark-client.service-account-registry get-config --username spark --namespace spark 
spark.eventLog.dir=s3a://spark-tutorial//spark

spark.eventLog.enabled=true

spark.executorEnv.LOKI_URL=http://grafana-agent-k8s-0.grafana-agent-k8s-endpoints.spark.svc.cluster.local:3500/loki/api/v1/push

spark.hadoop.fs.s3a.access.key=2SADHC75MH98XSKFA64C

spark.hadoop.fs.s3a.aws.credentials.provider=org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider

spark.hadoop.fs.s3a.connection.ssl.enabled=false

spark.hadoop.fs.s3a.endpoint=http://10.152.183.252

spark.hadoop.fs.s3a.path.style.access=true

spark.hadoop.fs.s3a.secret.key=sufsVr4X0ekcGHuS3Pi6HjN5MSAPGlTNoo0Ph4XY

spark.history.fs.logDirectory=s3a://spark-tutorial//spark

spark.kubernetes.driverEnv.LOKI_URL=http://grafana-agent-k8s-0.grafana-agent-k8s-endpoints.spark.svc.cluster.local:3500/loki/api/v1/push

spark.kubernetes.file.upload.path=s3a://spark-tutorial/

spark.sql.warehouse.dir=s3a://spark-tutorial/warehouse
spark.kubernetes.authenticate.driver.serviceAccountName=spark
spark.kubernetes.namespace=spark
```

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -29,17 +29,7 @@
...
-@pytest.fixture
-def namespace():
-    """A temporary K8S namespace gets cleaned up automatically."""
-    namespace_name = str(uuid.uuid4())
-    create_command = ["kubectl", "create", "namespace", namespace_name]
-    subprocess.run(create_command, check=True)
-    yield namespace_name
-    destroy_command = ["kubectl", "delete", "namespace", namespace_name]
-    subprocess.run(destroy_command, check=True)
+GRAFANA_AGENT_APP = "grafana-agent-k8s"
```
Comment (welpaolo) [inline tests/integration/test_charm.py:32]: Can you just add the `grafana-agent-k8s` in the `charm_versions` fixture?

**Review by welpaolo — COMMENTED** [review]
Looks good to me! I left a comments in the tests

**Review by deusebio — APPROVED** [review]
Changes are exactly what I would have expected! Thanks!

File: src/core/context.py [hunk 2]
```diff
@@ -130,6 +138,14 @@ def services_accounts(self) -> List[ServiceAccount]:
             if not relation or not relation.app
         ]
 
+    @property
+    def loki_url(self) -> str | None:
+        """Retrieve Loki URL form logging relations."""
```
Comment (theoctober19th) [inline src/core/context.py:143]:
```suggestion
        """Retrieve Loki URL from logging relations."""
```

**Review by theoctober19th — APPROVED** [review]

**Review by theoctober19th — APPROVED** [review]
LGTM, thanks!

**Review by theoctober19th — APPROVED** [review]

**Review by welpaolo — APPROVED** [review]
LGTM! Great contribution!

**Pull request merged**